server/channels/store/localcachelayer/post_layer.go

```diff
@@ -70,15 +70,15 @@ func (s LocalCachePostStore) InvalidateLastPostTimeCache(channelId string) {
 	}
 }
 
-func (s LocalCachePostStore) GetEtag(channelId string, allowFromCache, collapsedThreads bool) string {
+func (s LocalCachePostStore) GetEtag(channelId, userId string, allowFromCache, collapsedThreads bool) string {
 	if allowFromCache {
 		var lastTime int64
 		if err := s.rootStore.doStandardReadCache(s.rootStore.lastPostTimeCache, channelId, &lastTime); err == nil {
 			return fmt.Sprintf("%v.%v", model.CurrentVersion, lastTime)
 		}
 	}
 
-	result := s.PostStore.GetEtag(channelId, allowFromCache, collapsedThreads)
+	result := s.PostStore.GetEtag(channelId, userId, allowFromCache, collapsedThreads)
 
 	splittedResult := strings.Split(result, ".")
```

Comment:
This is now incorrect right? Your new logic has "%v.%v.%v", but we are saving only the last item in the cache.